src/platforms/web/compiler/modules/model.js

```diff
@@ -34,7 +34,7 @@ function preTransformNode (el: ASTElement, options: CompilerOptions) {
     if (map[':type'] || map['v-bind:type']) {
       typeBinding = getBindingAttr(el, 'type')
     }
-    if (!typeBinding && map['v-bind']) {
+    if (!map.type && !typeBinding && map['v-bind']) {
```

Comment:
This would make following case lose type binding: 
``` vue
<input type="checkbox" v-model="test" v-bind="{type: typeBindingValue}">
```

So maybe
```js
typeBinding = `(${map['v-bind']}).type || '${map.type}'`
```

Reply:
@hirokiosame If you could add the test as well, that would be great 👌

Reply:
I don't think we should allow `v-bind` to override a static `type`, it's gonna be confusing if done accidentally and I don't think there's a valid use case.